routing/router.go

```diff
@@ -1023,6 +1026,76 @@ func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *Route
 	return [32]byte{}, nil, sendError
 }
 
+// SendToRoute attempts to send a payment  through the passed
+// and defined by sender routes to the final destination.
+// This function is blocking and will return either: when
+// the passed routes are existed and the payment is successful, or all passed routes
+// have been attempted and resulted in a failed payment.
+// If the payment succeeds, then a non-nil Route
+// will be returned which describes the path the successful payment traversed
+// within the network to reach the destination. Additionally, the payment
+// preimage will also be returned.
+func (r *ChannelRouter) SendToRoute(payment *LightningPayment) ([32]byte, *Route, error) {
```

Comment:
I have done some optimizations and now the SandToRoute function is used by SendPayment function.